### ISSUE TITLE
Qt project: add dependency of _automoc target only when it's actually created

### ISSUE THREAD

**Issue by wdx04** [sakra/cotire#96]
Hi sakra,

When building Qt projects with CMake and cotire, I found CMake(the version I use is 3.5.2) will not create a separate _automoc target for small projects, even if AUTOMOC, AUTOUIC, AUTORCC are all set to ON. This will cause cotire to fail with a target-not-exist error. 
I suggest checking whether the _automoc target is created before the add_dependencies line:

```
    else()
        add_library(${_unityTargetName} ${_unityTargetSubType} EXCLUDE_FROM_ALL ${_unityTargetSources})
    endif()
-   if (_targetAutoMoc OR _targetAutoUic OR _targetAutoRcc)
+   if (TARGET ${_target}_automoc)
        # depend on the original target's implicity generated <targetname>_automoc target
        add_dependencies(${_unityTargetName} ${_target}_automoc)
    endif()
```

Thanks.

**Comment by sparsebase** [comment]
I have the same problem when using with CMake 3.5.0 to create Qt project, and wdx04's solution fixed it.

**Comment by Knitschi** [comment]
Had the same problem.

**Comment by jcelerier** [comment]
Damn, for me this fixes the problem on Windows with MSBuild, but then causes another problem on Linux : 

    make[3]: *** No rule to make target `foo_automoc.cpp', needed by `CMakeFiles/foo_unity.dir/foo_automoc.cpp.o'.

**Comment by jcelerier** [comment]
Had to do this to have it work everywhere...
https://github.com/OSSIA/i-score/commit/f198c07683508c1b42e6093498e8b0dc27da10b5#diff-3eeab8b40b7a1ecfb49292d650ff96bd

**Comment by sakra** [comment]
Fixed in 1.7.9.

**Comment by jcelerier** [comment]
thanks !
